playbook/Routing_IPv6_base_functionality/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml

```diff
@@ -0,0 +1,554 @@
+    with_items:
+        - port: "{{test_int1}}"
+          IPv6: "{{DUT1_IPV6_ADDRESS_1}}"
+        - port: "{{server_int}}"
+          IPv6: "{{DUT1_IPV6_ADDRESS_2}}"
+    register: ip_addresses
+    when: inventory_hostname == "DUT"
+  - name: TC3 - show IPv6 address
+    command: bash -c "ip addr show {{item}}"
+    with_items:
+            - "{{test_int1}}"
+            - "{{server_int}}"
+    register: show_ipv6_address
+    when: inventory_hostname == "DUT"
+  - name: TC3 - verify ipv6 address
+    assert:
+      that:
+        - '"inet6 {{DUT1_IPV6_ADDRESS_1}}" in show_ipv6_address.results[0].stdout_lines[2]'
+        - '"inet6 {{DUT1_IPV6_ADDRESS_2}}" in show_ipv6_address.results[1].stdout_lines[2]'
+      msg: "IPv6 setup failed on the ports"
+    when: inventory_hostname == "DUT"
+  - name: TC3 - set IPv6 address TR
+    command: "ip addr add {{TR1_IPV6_ADDRESS_1}}/{{MASK}} dev {{test_int1}}"
+    register: ip_addresses_tr
+    when: inventory_hostname == "TR"
+  - name: TC3 - show IPv6 address
+    command: bash -c "ip addr show {{test_int1}}"
+    register: show_ipv6_address_tr
+    when: inventory_hostname == "TR"
+  - name: TC3 - verify ipv6 address
+    assert:
+      that:
+        - '"inet6 {{TR1_IPV6_ADDRESS_1}}" in show_ipv6_address_tr.stdout_lines[2]'
+      msg: "IPv6 setup failed on the ports"
+    when: inventory_hostname == "TR"
+  - name: TC3 - Configure IPv6 Address - Server1
+    command: "ip addr add {{server1_ip}}/{{MASK}} dev {{server1_port}}"
+    register: ipv6_addr_server1
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Show IPv6 Address - Server1
+    command: bash -c "ip addr show dev {{server1_port}}"
+    register: show_ipv6_address_server1
+    when: inventory_hostname == "Server1"
+  - debug: msg="{{show_ipv6_address_server1.stdout}}"
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Verify IPv6 Address - Server1
+    assert:
+      that:
+        - '"inet6 {{server1_ip}}" in show_ipv6_address_server1.stdout_lines[2]'
+      msg: "IPv6 setup failed on the ports"
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Configure Default Route - Server1
+    command: bash -c "ip -6 route add default via {{DUT1_IPV6_ADDRESS_2}}"
+    register: dut_static_route
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Show IPv6 Route Configured - Server1
+    command: bash -c 'ip -6 route show | grep "{{DUT1_IPV6_ADDRESS_2}}"'
+    register: show_route
+    when: inventory_hostname == "Server1"
+  - debug: msg="{{show_route.stdout}}"
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Enable IPv6 Forwarding - DUT
+    command: bash -c 'echo "1" > /proc/sys/net/ipv6/conf/all/forwarding'
+    register: ipv6_forwarding_server1
+    when: inventory_hostname == "Server1"
+  - name: TC3 - Configure Route - TR
+    command: bash -c "{{config_route_via_cps}} create --vrf_name default --dst '{{server1_ip}}/{{MASK}}' --nh_addr '{{DUT1_IPV6_ADDRESS_1}}'"
```

Comment:
Cannot find where config_route_via_cps variable is defined?

Reply:
it is defined in the group_var/all file.

Reply:
thanks for adding that